fala/apps/adviser/views.py

```diff
@@ -37,3 +38,36 @@ def get(self, request, *args, **kwargs):
 
 class AccessibilityView(TemplateView):
     template_name = "adviser/accessibility-statement.html"
+
+
+class SearchView(ListView):
+    def get(self, request, *args, **kwargs):
+        self._form = AdviserSearchForm(data=request.GET or None)
+        self._data = self._form.search()
+        return super().get(self, request, *args, **kwargs)
+
+    def get_template_names(self):
+        if self._data:
+            template = "adviser/results.html"
+        else:
+            template = "adviser/search.html"
+        return [template]
+
+    def get_queryset(self):
+        return self._data.get("results", None)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self._data:
+            pages = LaaLaaPaginator(self._data["count"], 10, 3, self._form.current_page)
+            params = {
+                "postcode": self._form.cleaned_data["postcode"],
+                "name": self._form.cleaned_data["name"],
+            }
+            categories = list(map(lambda c: "categories=" + c, self._form.cleaned_data["categories"]))
+            context.update(
+                {"form": self._form, "data": self._data, "pages": pages, "params": params, "categories": categories}
+            )
+        else:
```

Comment:
I have a mild preference for doing the common bits outside of the conditional, and having the conditional only do what's different.
In this case that would look something like
```
context.update(form=self._form, data=self._data)  # this isn't different from passing them as a dict

if self._data:
    <snip>
    context.update(pages=pages, params=params, categories=categories)
```